substrafl/algorithms/pytorch/torch_newton_raphson_algo.py

```diff
@@ -249,30 +249,6 @@ def _local_train(
 
             self._update_gradients_and_hessian(loss, current_batch_size)
 
-    def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_path):
-        """Executes the following operations:
-
-            * Create the torch dataloader using the batch size given at the ``__init__`` of the class
-            * Sets the model to `eval` mode
-            * Returns the predictions
-
-        Args:
-            predict_dataset (torch.utils.data.Dataset): predict_dataset build from the x returned by the opener.
-        """
-        predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._batch_size)
-
-        self._model.eval()
-
-        predictions = torch.Tensor([])
-        with torch.inference_mode():
-            for x in predict_loader:
-                x = x.to(self._device)
-                predictions = torch.cat((predictions, self._model(x)), 0)
-
-        predictions = predictions.cpu().detach()
-
-        self._save_predictions(predictions, predictions_path)
-
```

Comment:
You cannot delete the overwrite of the local predict because, in the NR algorithm, the index generator is only defined for train task, and not for predict task. As the index generator will be set to None, it'll raise an error BatchSizeNotFoundError (cf [here](https://github.com/Substra/substrafl/blob/918c900c14c02cecb51f79551cd39262123c1fae/substrafl/algorithms/pytorch/torch_base_algo.py#L148))

The "slow" tests are not run on each PR. To see this impact you'll need to run `pytest tests/algorithms/pytorch/test_newton_raphson.py` locally.